### PR TITLE
feat(json-el): access variables by name instead of JSON path

### LINKED ISSUE #1844: Conditions access variables by name
Since the workflow instance payload is no single JSON document anymore, conditions should access the variables directly.

AT:
* a condition on a sequence flow access a variable value by its name instead of a JSON path expression (e.g. `foo` instead of `$.foo`)
* if the value is a JSON document then I can access nested fields by a dot-notation (e.g. `foo.bar`)

### PR DESCRIPTION
**Pull request by saig0**
* adjust JSON condition parser to access variables by name instead of JSON path expression

BREAKING CHANGE :warning: 
* the existing conditions on exclusive gateways must be changed (i.e. remove prefix `$.`)

closes #1844

### CODE REVIEW

File: json-el/src/main/scala/io/zeebe/msgpack/el/JsonObject.scala [hunk 1]
```diff
@@ -65,9 +66,9 @@ case class JsonString(value: DirectBuffer) extends JsonObject with JsonConstant
   token.setValue(value, 0, value.capacity())
 }
 
-case class JsonPath(value: String) extends JsonObject {
-  val query: JsonPathQuery = new JsonPathQueryCompiler().compile(value)
-  val variableName = query.getTopLevelVariable
+case class JsonPath(variableName: DirectBuffer, path: List[String]) extends JsonObject {
+  val jsonPath = "$." + (bufferAsString(variableName) :: path).mkString(".")
+  val query: JsonPathQuery = new JsonPathQueryCompiler().compile(jsonPath)
```
Comment (Zelldon) [inline json-el/src/main/scala/io/zeebe/msgpack/el/JsonObject.scala]: We still need the top level variable name from the json query at the in/out mappings right? Or can we remove it

File: json-el/src/main/scala/io/zeebe/msgpack/el/JsonCondition.scala [hunk 2]
```diff
@@ -17,58 +17,43 @@ package io.zeebe.msgpack.el
...
-case class Equal(x: JsonObject, y: JsonObject) extends Comparison
-
-case class NotEqual(x: JsonObject, y: JsonObject) extends Comparison
+case class Equal(x: JsonObject, y: JsonObject) extends JsonCondition with Comparison
 
-case class LessThan(x: JsonObject, y: JsonObject) extends Comparison
+case class NotEqual(x: JsonObject, y: JsonObject) extends JsonCondition with Comparison
 
-case class LessOrEqual(x: JsonObject, y: JsonObject) extends Comparison
+case class LessThan(x: JsonObject, y: JsonObject) extends JsonCondition with Comparison
```
Comment (Zelldon) [inline json-el/src/main/scala/io/zeebe/msgpack/el/JsonCondition.scala]: Why is Comparison and Operator not a JsonCondition anymore?

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Looks good, some small questions before merging.

File: json-el/src/main/scala/io/zeebe/msgpack/el/JsonObject.scala (hunk 1 above)
Comment (saig0) [reply json-el/src/main/scala/io/zeebe/msgpack/el/JsonObject.scala]: The `topLevelVariable` is still used for input/output mappings.

File: json-el/src/main/scala/io/zeebe/msgpack/el/JsonCondition.scala (hunk 2 above)
Comment (saig0) [reply json-el/src/main/scala/io/zeebe/msgpack/el/JsonCondition.scala]: Would be also possible. It is just more common that classes extend a sealed trait directly.

Comment (Zelldon) [reply json-el/src/main/scala/io/zeebe/msgpack/el/JsonCondition.scala]: Ok :)

**Review by ChrisKujawa — APPROVED** [review]
Bors r+

**Pull request merged**